internal/clientcache/cmd/daemon/status.go

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package daemon
+
+import (
+	"context"
+	stderr "errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/internal/clientcache/internal/daemon"
+	"github.com/hashicorp/boundary/internal/cmd/base"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/version"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+var errDaemonNotRunning = stderr.New("daemon is not running")
+
+var (
+	_ cli.Command             = (*AddTokenCommand)(nil)
+	_ cli.CommandAutocomplete = (*AddTokenCommand)(nil)
+)
+
+type StatusCommand struct {
+	*base.Command
+}
+
+func (c *StatusCommand) Synopsis() string {
+	return "Get the status information of the running boundary daemon"
+}
+
+func (c *StatusCommand) Help() string {
+	helpText := `
+Usage: boundary daemon status [options]
+
+  Get the status of the boundary daemon:
+
+      $ boundary daemon status
+
+  For a full list of examples, please see the documentation.
+
+` + c.Flags().Help()
+	return strings.TrimSpace(helpText)
+}
+
+func (c *StatusCommand) Flags() *base.FlagSets {
+	set := c.FlagSet(base.FlagSetClient | base.FlagSetOutputFormat)
+	return set
+}
+
+func (c *StatusCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *StatusCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *StatusCommand) Run(args []string) int {
+	ctx := c.Context
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		c.PrintCliError(err)
+		return base.CommandUserError
+	}
+
+	resp, err := c.Status(ctx)
+	if err != nil {
+		c.PrintCliError(err)
+		return base.CommandCliError
+	}
+
+	res := &daemon.StatusResult{}
+	apiErr, err := resp.Decode(res)
+	if err != nil {
+		c.PrintCliError(err)
+		return base.CommandCliError
+	}
+	if apiErr != nil {
+		c.PrintApiError(apiErr, "Error from daemon when getting the status")
+		return base.CommandApiError
+
+	}
+	switch base.Format(c.UI) {
+	case "json":
+		c.UI.Output(resp.Body.String())
+	default:
+		c.UI.Output(printStatusTable(res))
+	}
+	return base.CommandSuccess
+}
+
+func (c *StatusCommand) Status(ctx context.Context) (*api.Response, error) {
+	const op = "daemon.(StatusCommand).Status"
+	dotPath, err := DefaultDotDirectory(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return status(ctx, dotPath)
+}
+
+func status(ctx context.Context, daemonPath string) (*api.Response, error) {
+	const op = "daemon.status"
+	client, err := api.NewClient(nil)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
```

Comment:
These errors will bubble up to printing directly to CLI. We don't return raw errors for CLI commands; they are human readable, capitalized with punctuation (but can also include raw error strings). For example: `c.PrintCliError(fmt.Errorf("Error creating API client: %w", err))`

Reply:
Also we don't include the op anywhere in the CLI at this point. Maybe we should but I don't think we should introduce that pattern here.